lisa/src/main/java/it/unive/lisa/program/CompilationUnit.java

```diff
@@ -589,6 +595,46 @@ else if (!overriding.isEmpty())
 					}
 			}
 
+		for (CompilationUnit superUnit : superUnits)
+			for (Annotation ann : superUnit.getAnnotations()) {
+				Annotation inheritedAnn = new Annotation(ann.getAnnotationName(), ann.getAnnotationMembers(), true);
+				addAnnotation(inheritedAnn);
+			}
+
+		for (CodeMember instCfg : getInstanceCodeMembers(false))
+			for (CodeMember matching : instCfg.getDescriptor().overrides())
+				for (Annotation ann : matching.getDescriptor().getAnnotations()) {
+					Annotation inheritedAnn = new Annotation(ann.getAnnotationName(), ann.getAnnotationMembers(), true);
+					instCfg.getDescriptor().addAnnotation(inheritedAnn);
+
+					Parameter[] args = instCfg.getDescriptor().getArgs();
+					Parameter[] superArgs = matching.getDescriptor().getArgs();
+					for (int i = 0; i < args.length; i++)
+						for (Annotation parAnn : superArgs[i].getAnnotations()) {
+							Annotation inheritedParAnn = new Annotation(parAnn.getAnnotationName(),
+									parAnn.getAnnotationMembers(), true);
+							args[i].addAnnotation(inheritedParAnn);
+						}
+				}
+
 		hierarchyComputed = true;
 	}
+
+	/**
+	 * Yields the annotations of this compilation unit.
+	 * 
+	 * @return the annotations of this compilation unit
+	 */
+	public Annotations getAnnotations() {
+		return annotations;
+	}
+
+	/**
+	 * Adds an annotation to the annotations of this compilation unit.
+	 * 
+	 * @param ann the annotation to be added
+	 */
+	public void addAnnotation(Annotation ann) {
```

Comment:
In some classes we have `addAnnotation`, in some other we have `setAnnotations`. We should unify them